docs/core/tutorials/netcore-hosting.md

```diff
@@ -0,0 +1,312 @@
+
+This tutorial and its [associated sample](https://github.com/dotnet/docs/tree/master/samples/core/hosting) build a Windows host, but please see the notes at the end of this article about hosting on Unix.
+
+## Creating the host
+
+A sample host demonstrating the steps outlined in this article is available in our [.NET Core samples](https://github.com/dotnet/docs/tree/master/samples/core/hosting). Comments in the sample's host.cpp file clearly associate the numbered steps from this tutorial with where they are performed in the sample.
+
+Please bear in mind that the sample host is meant to be used for learning purposes so it is light on error checking and bears a design that emphasizes readability over efficiency. More real-world host samples are available in the [dotnet/coreclr](https://github.com/dotnet/coreclr/tree/master/src/coreclr/hosts) repository. The [CoreRun host](https://github.com/dotnet/coreclr/tree/master/src/coreclr/hosts/corerun), in particular, is a good general-purpose host to study after reading through the simpler sample.
+
+### A note about mscoree.h
+The primary .NET Core hosting interface (`ICLRRuntimeHost2`) is defined in [MSCOREE.IDL](https://github.com/dotnet/coreclr/blob/master/src/inc/MSCOREE.IDL). A header version of this file (mscoree.h - which your host will need to reference) is produced via MIDL when the [.NET Core runtime](https://github.com/dotnet/coreclr/) is built. If you do not want to build the .NET Core runtime, mscoree.h is also available as a [pre-built header](https://github.com/dotnet/coreclr/tree/master/src/pal/prebuilt/inc) in the dotnet/coreclr repository. 
+
+### Step 1 - Identify the managed entry point
+After referencing necessary headers ([mscoree.h](https://github.com/dotnet/coreclr/tree/master/src/pal/prebuilt/inc/mscoree.h) and stdio.h, for example), one of the first things a .NET Core host must do is locate the managed entry point it will be using.
+
+In our sample host, this is done very simply - we just take the first argument to our host as the path to a managed binary whose main method will be executed.
+
+```C++
+// The managed application to run should be the first command line parameter.
+// Subsequent command line parameters will be passed to the managed app later in this host.
+wchar_t targetApp[MAX_PATH];
+GetFullPathNameW(argv[1], MAX_PATH, targetApp, NULL);
+```
+
+### Step 2 - Find and load CoreCLR.dll
+The .NET Core runtime APIs are in CoreCLR.dll (on Windows). To get our hosting interface (`ICLRRuntimeHost2`), it's necessary to find and load CoreCLR.dll. It is up to the host to define a convention for how it will locate CoreCLR.dll. Some hosts expect the file to be present in a well-known machine-wide location (such as %programfiles%\dotnet\shared\Microsoft.NETCore.App\1.1.0). Others expect that CoreCLR.dll will be loaded from a location next to either the host itself or the app to be hosted. Still others might consult an environment variable to find the library.
```

Comment:
We should italicize the file names. See https://github.com/dotnet/docs/blob/master/styleguide/template.md#text-styling for more info, if needed